### ISSUE TITLE
Python Scanner generates False Positives

### ISSUE THREAD

**Issue by san-zrl** [PQCA/sonar-cryptography#300]
When analysing https://github.com/docling-project/docling the python scanner creates FPs on code related to model generation such as 

``` python
generated_ids = self.vlm_model.generate(
    **inputs,
    max_new_tokens=self.max_new_tokens,
    use_cache=self.use_cache,
    temperature=self.temperature,
```

This finding comes across as 

Asset Type:	Algorithm
Curve:	Edwards448
Crypto Functions:	Key Pair Generation
OID:	1.3.101.113

**Comment by Arijit429** [comment]
Opened a PR that addresses this. The generate() rules for Ed25519 and Ed448 
were using withAnyParameters(), which matches any .generate() call regardless 
of parameters. Since Ed25519PrivateKey.generate() and Ed448PrivateKey.generate() 
take no parameters, changed both to withoutParameters() in PycaSign.java.
